File: 1323.Maximum 69 Number.cc
// https://leetcode.com/problems/maximum-69-number/submissions/

class Solution {
public:
    int maximum69Number (int n) {
        
        int exp = floor(log10(n)); // exp represents power of 10 - number of digits of input num
        int maximum = 0;
        int div = pow(10, exp);
        
        while(exp >= 0) {
            div = pow(10, exp);
            
            if(n / div == 6 ) {
                maximum += 9 * div;
                maximum += (n % div); 
                break;
            }

            maximum += n - (n % div);

            n %= div;
            --exp;
        }

        return maximum;
    }
};

File: leetcode/1201-1500/1301-1350/1323.Maximum 69 Number.rs
// https://leetcode.com/problems/maximum-69-number

impl Solution {
    pub fn maximum69_number (num: i32) -> i32 {
        let mut max = num;
        let mut div = 10000;
        while div > 0 {
            if (max % (div * 10) / div) == 6 {
                return num + (div * 3)
            }
            div /= 10
        }
        max
    }
}